### PR TITLE
chore(ci): create full benchmarks workflow that runs weekly

### PR DESCRIPTION
**Pull request by soonum**

### CODE REVIEW

**Review by IceTDrinker — COMMENTED** [review]
Main question is why not re-use the existing benchmark files ?

I think the only benchmark needing an adaptation or a dedicated workflow are shortint (to make things simpler) and integer + integer multi bit ?

**Comment by IceTDrinker** [comment]
@slab-ci cpu_fast_test

**Comment by soonum** [comment]
> Main question is why not re-use the existing benchmark files ?
> 
> I think the only benchmark needing an adaptation or a dedicated workflow are shortint (to make things simpler) and integer + integer multi bit ?

Yes you're right but it shortint and integer adaptation would require careful thinking to be able to handle both case (fast bench VS full bench)

**Comment by IceTDrinker** [comment]
> > Main question is why not re-use the existing benchmark files ?
> > I think the only benchmark needing an adaptation or a dedicated workflow are shortint (to make things simpler) and integer + integer multi bit ?
> 
> Yes you're right but it shortint and integer adaptation would require careful thinking to be able to handle both case (fast bench VS full bench)

worst case we can have a dedicated full bench one if it's too annoying to adapt, separate workflows have the advantage to parallelize the benchmarks (even if they run during the week-end)

**Comment by soonum** [comment]
Here what I propose, keeping the new file `start_full_benchmark.yml`. Inside it we'll call the exisiting short benchmarks (boolean, pbs and wasm). In addition to that, I'd break the `full_benchmark.yml` into two separate files `full_shortint_benchmarks.yml` and `full_integer_benchmarks.yml`.
That way we could benefit from parallelism, with not too much duplication. Moreover, handling matrix benchmarks into existing workflows for shortint and integer would be a nightmare.

What do you think about that design?

**Comment by IceTDrinker** [comment]
@soonum perfect 🙂 that's what I had in mind as well !

**Comment by IceTDrinker** [comment]
one last detail it feels like it would be _super_ expensive to run, maybe let's not enable the schedule right now until we have one full run to assess the runtime and cost wdyt ?

File: .github/workflows/integer_full_benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  integer-benchmarks:
+    name: Execute integer benchmarks for all operations flavor
+    runs-on: ${{ github.event.inputs.runner_name }}
+    if: ${{ !cancelled() }}
+    needs: prepare-benchmarks
+    strategy:
+      max-parallel: 1
+      matrix:
+        command: [ integer, integer_multi_bit]
+        op_flavor: [ default, default_scalar, smart, smart_scalar, smart_parallelized, smart_scalar_parallelized, unchecked, unchecked_scalar, misc ]
```
Comment (IceTDrinker) [inline .github/workflows/integer_full_benchmark.yml:79]: 18 combinations in total is that right ? single bench machine with up to 6 hours per bench

108 hours of bench ? 😱

**Review by IceTDrinker — APPROVED** [review]
Looks good but I'm worried for bench timings

File: .github/workflows/integer_full_benchmark.yml (hunk 1 above)
Comment (soonum) [reply .github/workflows/integer_full_benchmark.yml:79]: Yes I know but we'll see what is the actual number.

**Pull request merged**